src/dataflow/source/kafka.rs

```diff
@@ -178,12 +242,21 @@ where
                 while let Some(message) = next_message {
                     let partition = message.partition;
                     let offset = message.offset + 1;
-                    if partition_metadata.len() <= partition as usize {
```

Comment:
It seems like you're forgetting to buffer the message here?

Reply:
Fixed, thanks.